### ISSUE TITLE
Module not found: Error: Cannot resolve module 'child_process' in ..socket.io

### ISSUE THREAD

**Issue by jeberly** [webpack/webpack-dev-server#71]
I am new to webpack, I was following this tutorial => http://webpack.github.io/docs/tutorials/getting-started/

Everything worked great until I got to the last section titled "DEVELOPMENT SERVER".  After following the instructions, I get the following output (along with more similar errors) when visiting. 

http://localhost:8080/webpack-dev-server/

Errors while compiling. App updated with errors. No reload!

(webpack)-dev-server/~/socket.io/lib/static.js
Module not found: Error: Cannot resolve module 'child_process' in /Users/jeberly/work/mydomain.com/webpack/node_modules/webpack-dev-server/node_modules/socket.io/lib
resolve module child_process in /Users/jeberly/work/mydomain.com/webpack/node_modules/webpack-dev-server/node_modules/socket.io/lib

**Comment by omerts** [comment]
trying adding this to your config:
```javascript
node: {
      child_process: 'empty'
    }
```

**Comment by SpaceK33z** [comment]
@jeberly, we don't use socket.io anymore. Do you still have this issue?

**Comment by SpaceK33z** [comment]
I'm closing this issue because of inactivity. Feel free to comment, and we can always re-open it again.

**Comment by jogshraddha** [comment]
I am facing same issue when I have added node serialport package in my application

> _Module not found: Error: Cannot resolve module 'child_process' in /project/node_modules/serialport/lib/bindings

My application is created using create-react-app and electronjs. I googled it and the solutions mostly suggested are related to webpack config which I can not directly modify in create-react-app. I tried those solutions like adding 'target: node' and external config etc by ejecting create react app but that also did not work. 
Befoe adding serialport package it was working properly now it looks up for the child_process package into serialport libs. 
Can someone help me out in this?
